src/conf_mode/protocols_rip.py

```diff
@@ -0,0 +1,313 @@
+            }
+        })
+
+    # Get passive-interface
+    if conf.exists_effective('passive-interface'):
+        rip_conf['old_rip']['passive_iface'] = conf.return_effective_values('passive-interface')
+
+    if conf.exists('passive-interface'):
+        rip_conf['rip']['passive_iface'] = conf.return_values('passive-interface')
+
+    # Get redistribute for old_rip
+    for protocol in conf.list_effective_nodes('redistribute'):
+        rip_conf['old_rip']['redist'].update({
+            protocol : {
+                'metric' : conf.return_effective_value('redistribute {0} metric'.format(protocol)),
+                'route_map' : conf.return_effective_value('redistribute {0} route-map'.format(protocol)),
+            }
+        })
+
+    # Get redistribute
+    for protocol in conf.list_nodes('redistribute'):
+        rip_conf['rip']['redist'].update({
+            protocol : {
+                'metric' : conf.return_value('redistribute {0} metric'.format(protocol)),
+                'route_map' : conf.return_value('redistribute {0} route-map'.format(protocol)),
+            }
+        })
+
+    conf.set_level('protocols rip')
+
+    # Get route
+    if conf.exists_effective('route'):
+        rip_conf['old_rip']['route'] = conf.return_effective_values('route')
+
+    if conf.exists('route'):
+        rip_conf['rip']['route'] = conf.return_values('route')
+
+    # Get timers garbage
+    if conf.exists_effective('timers garbage-collection'):
+        rip_conf['old_rip']['timer_garbage'] = conf.return_effective_value('timers garbage-collection')
+
+    if conf.exists('timers garbage-collection'):
+        rip_conf['rip']['timer_garbage'] = conf.return_value('timers garbage-collection')
+
+    # Get timers timeout
+    if conf.exists_effective('timers timeout'):
+        rip_conf['old_rip']['timer_timeout'] = conf.return_effective_value('timers timeout')
+
+    if conf.exists('timers timeout'):
+        rip_conf['rip']['timer_timeout'] = conf.return_value('timers timeout')
+
+    # Get timers update
+    if conf.exists_effective('timers update'):
+        rip_conf['old_rip']['timer_update'] = conf.return_effective_value('timers update')
+
+    if conf.exists('timers update'):
+        rip_conf['rip']['timer_update'] = conf.return_value('timers update')
+
+    return rip_conf
+
+def verify(rip):
+    if rip is None:
+        return None
+
+    # Check for network. If network-distance acl is set and distance not set
+    for net in rip['rip']['net_distance']:
+        if not rip['rip']['net_distance'][net]['distance']:
+            raise ConfigError(f"Must specify distance for network {net}")
+
+def generate(rip):
+    if rip is None:
+        return None
+
+    render(config_file, 'frr/rip.frr.tmpl', rip)
+    return None
+
+def apply(rip):
+    if rip is None:
+        return None
+
+    if os.path.exists(config_file):
+        call("sudo vtysh -d ripd -f " + config_file)
+        os.remove(config_file)
```

Comment:
You should always check if a file exists to avoid any exception.

Reply:
done
```
    else:
        print("File {0} not found".format(config_file))
```